src/servers/src/http/event.rs

```diff
@@ -147,6 +147,51 @@ where
     }
 }
 
+#[axum_macros::debug_handler]
+pub async fn query_pipeline(
+    State(state): State<LogState>,
+    Extension(mut query_ctx): Extension<QueryContext>,
+    Query(query_params): Query<LogIngesterQueryParams>,
+    Path(pipeline_name): Path<String>,
+) -> Result<GreptimedbManageResponse> {
+    let start = Instant::now();
+    let handler = state.log_handler;
+    ensure!(
+        !pipeline_name.is_empty(),
+        InvalidParameterSnafu {
+            reason: "pipeline_name is required in path",
+        }
+    );
+
+    let version_str = query_params.version.context(InvalidParameterSnafu {
```

Comment:
`version` can be optional. If the user don't specify the version, you can use the latest version.